client/src/components/UserManagement.jsx

```diff
@@ -0,0 +1,201 @@
+    headerName: "Email",
+    width: 535,
+    editable: false,
+    sortable: false,
+    filterable: false,
+  },
+  {
+    field: "actions",
+    type: "actions",
+    headerName: "",
+    width: 160,
+    hide: true,
+    cellClassName: "actions",
+    getActions: () => {
+      return [
+        <GridActionsCellItem
+          icon={<EditIcon />}
+          label="Edit"
+          className="textPrimary actionButton"
+          // onClick={handleEditClick(id)}
+          color="inherit"
+        />,
+        <GridActionsCellItem
+          icon={<DeleteIcon />}
+          label="Delete"
+          color="inherit"
+          className="actionButton"
+        />,
+      ];
+    },
+  },
+];
+
+const rows = [
+  { id: 1, name: "First Last", email: "email@email.com" },
+  { id: 2, name: "First Last", email: "email@email.com" },
+  { id: 3, name: "First Last", email: "email@email.com" },
+  { id: 4, name: "First Last", email: "email@email.com" },
+  { id: 5, name: "First Last", email: "email@email.com" },
+  { id: 6, name: "First Last", email: "email@email.com" },
+  { id: 7, name: "First Last", email: "email@email.com" },
+  { id: 8, name: "First Last", email: "email@email.com" },
+  { id: 9, name: "First Last", email: "email@email.com" },
+  { id: 10, name: "First Last", email: "email@email.com" },
+];
+
+export default function DataGridDemo() {
+  const [searchQuery, setSearchQuery] = React.useState("");
+  const [filteredRows, setFilteredRows] = React.useState(rows);
+
+  const handleSearch = (event) => {
+    const query = event.target.value;
+    setSearchQuery(query);
+
+    const filtered = rows.filter((row) =>
+      row.name.toLowerCase().includes(query.toLowerCase()),
+    );
+    setFilteredRows(filtered);
+  };
+
+  const handleFilterReset = () => {
+    setSearchQuery("");
+    setFilteredRows(rows);
+  };
+
+  return (
+    <div style={{ marginRight: 40 }}>
+      <div style={{ display: "flex", alignItems: "center" }}>
+        <ArrowBackIcon sx={{ color: "gray", marginRight: 2, marginLeft: 2 }} />
+        <Typography
+          style={{
+            fontFamily: "Arial",
+            fontSize: "34px",
+            fontWeight: 500,
+            lineHeight: "42px",
+            letterSpacing: "0.25px",
+            textAlign: "left",
+          }}
+        >
+          Admin Dashboard
+        </Typography>
+        <Button
+          sx={{
+            marginLeft: "auto",
+            marginBottom: "30px",
+            marginTop: "30px",
+            backgroundColor: "#3568E5",
+            color: "white",
+            "&:hover": {
+              backgroundColor: "#3568E5",
+              color: "white",
+            },
+          }}
+          startIcon={<AddIcon />}
+        >
+          New Users
+        </Button>
+      </div>
+
+      <Box sx={{ height: 400, width: "100%" }}>
+        <div
+          style={{
+            display: "flex",
+            flexDirection: "row",
+            // display: "grid",
```

Comment:
Please delete any comments that aren't there to help other devs understand your code